registry/reference.go

```diff
@@ -114,6 +120,9 @@ func ParseReference(artifact string) (Reference, error) {
 		// `tag` found; Valid Form C
 		repository = path[:index]
 		reference = path[index+1:]
+		if !tagRegexp.MatchString(reference) {
+			return Reference{}, errdef.ErrInvalidReferenceTerminal
```

Comment:
Here we have a digest that snuck into this code block under the pretense of being a tag.